include/chef_types.hrl

```diff
@@ -219,15 +220,20 @@
 %% Not a true chef object either, keys are owned by their associated user
 %% or client object and are managed in that context.
 -record(chef_key, {
-	  'id'  :: object_id(),     %% guid of user or client, unique with key_name
-	  'key_name',               %% user named string describing key
-	  'public_key',             %% PKCS#1 public key or a certificate
-	  'key_version',            %% 0 for public_key, 1 for cert
-	  'expires_at' :: binary(), %% expiration time (utc)
-	  'last_updated_by' :: binary(), %% authz id of updater
-	  'created_at' :: binary(), %% when created
-	  'updated_at' :: binary()  %% when last updated
-	 }).
+          'id'  :: object_id(),     %% guid of user or client, unique with key_name
+          'key_name',               %% user named string describing key
+          'public_key',             %% PKCS#1 public key or a certificate
+          'key_version',            %% 0 for public_key, 1 for cert
+          'expires_at' :: binary(), %% expiration time (utc)
+          'last_updated_by' :: binary(), %% authz id of updater
+          'created_at' :: binary(), %% when created
+          'updated_at' :: binary(),  %% when last updated
+
+          %% Important note - in 'chef_key:flatten' we explcitly drop this last field
```

Comment:
This matches what we do in other places I believe, so this comment shoudn't block merging.  However, rather than relying on flatten, I wonder if we should user fields_for_insert or a new call back that more explicitly constructs the fields for DB insertion.  We ended up doing something like that here:

https://github.com/chef/oc_erchef/commit/0c07d1b434481d045aad9ca67a6a07be33da8979

when trying to maintain the record for insert based on the ordering of the record got a bit crazy.